<a href="https://colab.research.google.com/github/Frankr22/supervised-machine-learning-challenge/blob/main/Credit_Risk_Evaluator.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Credit Risk Evaluator

In [1]:
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split

## Retrieve the Data

The data is located in the Challenge Files Folder:

* `lending_data.csv`

Import the data using Pandas. Display the resulting dataframe to confirm the import was successful.

In [2]:
# Import the data
# Load data from CSV file
lending_data = pd.read_csv("Resources/lending_data.csv")

# Display the dataframe
print(lending_data.head())

   loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0  


## Predict Model Performance

You will be creating and comparing two models on this data: a Logistic Regression, and a Random Forests Classifier. Before you create, fit, and score the models, make a prediction as to which model you think will perform better. You do not need to be correct! 

Write down your prediction in the designated cells in your Jupyter Notebook, and provide justification for your educated guess.

My prediction is that the Random Forest Classifier might perform better in this case, as there could be non-linear relationships between the features that affect the loan status.

## Split the Data into Training and Testing Sets

In [3]:
# Split the data into X_train, X_test, y_train, y_test

# Split the data into features (X) and labels (y)
X = lending_data.drop("loan_status", axis=1)
y = lending_data["loan_status"]

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=1)

## Create, Fit and Compare Models

Create a Logistic Regression model, fit it to the data, and print the model's score. Do the same for a Random Forest Classifier. You may choose any starting hyperparameters you like. 

Which model performed better? How does that compare to your prediction? Write down your results and thoughts in the designated markdown cell.

In [6]:
# Train a Logistic Regression model and print the model score
from sklearn.linear_model import LogisticRegression

# Create a Logistic Regression model
lr_model = LogisticRegression(solver='lbfgs', max_iter=1000)

# Fit the model to the training data
lr_model.fit(X_train, y_train)

# Print the model score on the testing data
print("Logistic Regression model score: {:.3f}".format(lr_model.score(X_test, y_test)))

Logistic Regression model score: 0.992


In [7]:
# Train a Random Forest Classifier model and print the model score
from sklearn.ensemble import RandomForestClassifier

# Create a Random Forest Classifier model
rf_model = RandomForestClassifier(n_estimators=100, max_depth=None, random_state=1)

# Fit the model to the training data
rf_model.fit(X_train, y_train)

# Print the model score on the testing data
print("Random Forest Classifier model score: {:.3f}".format(rf_model.score(X_test, y_test)))

Random Forest Classifier model score: 0.991


The Logistic Regression model performed slightly better than the Random Forest Classifier model, with a score of 0.992 compared to 0.991 for the Random Forest Classifier model. This result is contrary to my previous prediction, as the Logistic Regression model would typically perform better when the features are linearly separable, which appears to be the case in this dataset.